### PR TITLE
Added Detecting and Removing Cycle from a linked list 

### LINKED ISSUE #520: Linked List Trickiest Questions in C++
Hey...I want to add trickiest and mind boggling questions on linked list in C++.
For example - Converting singly linked list into circular linked list.
                       Removing loop from a linked list
                       Partitioning linked list around a given value
and lots more.

### PR DESCRIPTION
**Pull request by saumyashrm2802**
Fixes #520 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added C++ code for Detecting and removing Cycle from a linked list
(will add code for more trickiest questions soon)
 

#### Languages Used:
- C++

#### Files Added:
- Detecting and Removing Loop from ll.cpp

### CODE REVIEW

File: Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp [hunk 1]
```diff
@@ -1,78 +1,98 @@
...
+
 struct node
 {
-    int info;
-    node *next;
+  int info;
+  node *next;
 };
-node * create(int x)
+node *create (int x) //creating a new node
```
Comment (somya-kapoor) [inline Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp]: Please add a new line above it

File: Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp [hunk 2]
```diff
@@ -1,78 +1,98 @@
...
 {
-    node *p =new node;
-    p->info=x;
-    p->next=NULL;
-    return p;
+  node *p = new node;
+  p->info = x;
+  p->next = NULL;
+  return p;
```
Comment (somya-kapoor) [inline Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp]: Remove extra line

File: Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp [hunk 3]
```diff
@@ -1,78 +1,98 @@
...
+/* Sample input :HARDCORED VALUES:
+  10->20->30->(attached to head)
+
+   Sample output:
+   A cycle is present in the linked list
+
+   After removing the cycle, linked list is as follows:
+   10 20 30
+/*
+
```
Comment (somya-kapoor) [inline Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp]: Remove the extra lines

**Comment by somya-kapoor** [comment]
Please fix indentation in while and if statements.

**Comment by saumyashrm2802** [comment]
@somya-kapoor  I have made the required changes. Kindly have a look

**Comment by somya-kapoor** [comment]
@saumyashrm2803 You haven't fixed indentation in while and if statement.

File: Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp [hunk 4]
```diff
@@ -1,78 +1,98 @@
...
+  return p;
 
 }
-void print (node *ptr)
+
+void print (node * ptr) //printing the linked list
 {
-    while(ptr!=NULL)
+  while (ptr != NULL)
     {
```
Comment (somya-kapoor) [inline Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp]: Fix indentation of while statement.
It should be:
while
{

File: Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp [hunk 5]
```diff
@@ -1,78 +1,98 @@
...
-    node *fast=ptr;
-    while(slow && fast && fast->next)
+  node *slow = ptr;
+  node *fast = ptr;
+  while (slow && fast && fast->next)
     {
-        slow=slow->next;
-        fast=fast->next->next;
-        if(slow==fast)
-            {rem(slow,ptr);
```
Comment (somya-kapoor) [inline Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp]: Same goes with if.
if
{

File: Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp [hunk 6]
```diff
@@ -1,78 +1,98 @@
...
+      while (temp->next != slow && temp->next != ptr)
+	{
+	  temp = temp->next;
+	}
+      if (temp->next == ptr)
+	break;
+      ptr = ptr->next;
     }
-    temp->next=NULL;
+  temp->next = NULL;
```
Comment (somya-kapoor) [inline Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp]: Please remove extra line

File: Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp (hunk 4 above)
Comment (somya-kapoor) [reply Linked list Trickiest Questions/Detecting and Removing loop from ll.cpp]: Also, have the comments about the function, above its definition for better readability. Try following this with other general comments too.

**Comment by jainaman224** [comment]
@saumyashrm2802 Please fix file name.

File: Linked list Trickiest Questions/Detecting and Removing loop from Linked List.cpp [hunk 7]
```diff
@@ -0,0 +1,99 @@
+#include<iostream>
+#include<conio.h>
+using namespace std;
+
+struct node
+{
+  int info;
```
Comment (SakshiShreya) [inline Linked list Trickiest Questions/Detecting and Removing loop from Linked List.cpp]: Use 4 space indentation

Comment (saumyashrm2802) [reply Linked list Trickiest Questions/Detecting and Removing loop from Linked List.cpp]: @SakshiShreya Mam may you please explain the way to indent the code using 4 spaces. And do I have to make this change to the entire code or to only these four lines that you just mentioned?

Comment (marwahmanbir) [reply Linked list Trickiest Questions/Detecting and Removing loop from Linked List.cpp]: @saumyashrm2802 you have to add space indentation in entire code. You can convert tab indentation to space indentation right from the text editor. If you're using Sublime/VSCode, do the following:
- ctrl+shift+P
- Search for "indentation"
- Select "Convert tabs to spaces" or "Indentation to Spaces"
- Save the file and commit

Comment (saumyashrm2802) [reply Linked list Trickiest Questions/Detecting and Removing loop from Linked List.cpp]: @marwahmanbir Thankyou so much for the clarification. I have corrected the indentation.

**Comment by saumyashrm2802** [comment]
@SakshiShreya  
I am sorry I did a wrong commit with message "Final". I was trying to squash all commits but something else happened. I think I should create another branch and submit a new pull request containing the desired file.

**Pull request closed without merging**